--- a/drivers/gpu/host1x-fence/dev.c
+++ b/drivers/gpu/host1x-fence/dev.c
@@ -202,6 +202,7 @@
 {
 	struct host1x_pollfd *pollfd = file->private_data;
 	struct host1x_pollfd_fence *pfd_fence, *pfd_fence_temp;
+	unsigned long irqflags;
 
 	mutex_lock(&pollfd->lock);
 
@@ -212,8 +213,8 @@
 			pfd_fence->callback_set = false;
 		}
 		/*The lock/unlock just ensures that the callback execution has finished*/
-		spin_lock(pfd_fence->fence->lock);
-		spin_unlock(pfd_fence->fence->lock);
+		spin_lock_irqsave(pfd_fence->fence->lock, irqflags);
+		spin_unlock_irqrestore(pfd_fence->fence->lock, irqflags);
 
 		dma_fence_put(pfd_fence->fence);
 		kfree(pfd_fence);
@@ -231,6 +232,7 @@
 	struct host1x_pollfd *pollfd = file->private_data;
 	struct host1x_pollfd_fence *pfd_fence, *pfd_fence_temp;
 	unsigned int mask = 0;
+	unsigned long irqflags;
 
 	poll_wait(file, &pollfd->wq, wait);
 
@@ -247,8 +249,8 @@
 				pfd_fence->callback_set = false;
 			}
 			/*The lock/unlock just ensures that the callback execution has finished*/
-			spin_lock(pfd_fence->fence->lock);
-			spin_unlock(pfd_fence->fence->lock);
+			spin_lock_irqsave(pfd_fence->fence->lock, irqflags);
+			spin_unlock_irqrestore(pfd_fence->fence->lock, irqflags);
 
 			dma_fence_put(pfd_fence->fence);
 			list_del(&pfd_fence->list);
